core/src/test/resources/HistoryServerExpectations/stage_with_speculation_summary_expectation.json

```diff
@@ -0,0 +1,748 @@
+{
```

Comment:
This change includes over 700 lines of code. Can we minimize this?

Reply:
since the speculation summary is part of stage api which is why we see the whole output, not sure how I can minimize this further. I think already this app only ran 10 tasks only.

Reply:
Could you reproduce speculation with 2 or 3 tasks?

Reply:
Do you see a big concern here? The thing is with this change then I have to make changes all those expectation files again. Let me know if you think it could be a issue, will try to fix it.

Reply:
It just seems redundant. If it's difficult to reduce it, we can replace it later with a small one.

Reply:
Will try to reduce it over the weekend.

Reply:
@sarutak I tried minimizing this file, it is still 500 odd lines. I don't think this can be reduced further. Ran this with only 4 tasks. Earlier we had a separate API for `speculationSummary` if we go with that approach then this expectation file size can be reduced. Let me know your thoughts.